containers/src/Kokkos_UnorderedMap.hpp

```diff
@@ -526,7 +526,8 @@ class UnorderedMap {
   ///                       Kokkos::UnorderedMapInsertOpTypes for more ops.
   template <typename InsertOpType = default_op_type>
   KOKKOS_INLINE_FUNCTION insert_result
-  insert(key_type const &k, impl_value_type const &v = impl_value_type(),
+  insert(key_type const &k,
+         [[maybe_unused]] impl_value_type const &v   = impl_value_type(),
```

Comment:
The `[[maybe_unused]]` attribute on arguments of a function of a function whose implementation does not fit on one page is an example of "code smell", that is to say I am not convinced since is an improvement whatsoever.

Reply:
I was "forced" by the compiler to add `[[maybe_unused]]` because this PR guards some parts of the `insert` code with `if constexpr` (instead of `if`). Therefore, compiling with a `void` value type (*i.e.* making a set instead of a map) makes these code blocks guarded by `if constexpr` not instantiated, and so the compiler complains "uho you have `v` argument that is unused` !"...

Reply:
(So yes, the addition of `constexpr` is not just a useless change, it actually changes how compilers behave - for good in this case IMO since it's now clearly written that `v` might be unused, and that's the case for a set)

Reply:
I understand what made you add the attribute.  I am just saying that it is not clear to me that the code is better (as in more readable, more maintainable) now than before.